### PR TITLE
[Master-4.1] Add option for fail on undeclared dependency error to maven plugin

### PR DESCRIPTION
**Pull request by fwellers**

### CODE REVIEW

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java [hunk 1]
```diff
@@ -52,6 +55,14 @@
   @Parameter(property = "maven.compiler.failOnError")
   private boolean failOnError = true;
 
+  /**
+   * Indicates whether the build will fail if there are dependency errors discovered.
+   * Defaults to "false".
```
Comment (fwienber) [inline jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]:
```suggestion
   * Defaults to "true".
```

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java [hunk 2]
```diff
@@ -52,6 +55,14 @@
   @Parameter(property = "maven.compiler.failOnError")
   private boolean failOnError = true;
 
+  /**
+   * Indicates whether the build will fail if there are dependency errors discovered.
```
Comment (fwienber) [inline jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java:59]: I think the build only fails if _used, undeclared_ dependencies are discovered, so the documentation should say so.

Comment (fwienber) [reply jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java:59]: In other words, it should be elaborated what a dependency _error_ is. Currently, we would consider unused dependencies a warning, but used, undeclared dependencies an error.

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java [hunk 3]
```diff
@@ -179,11 +190,11 @@
    * In ECMAScript, initializer values are assigned to all 'undefined' arguments.
    * In AS3, initializer values are assigned only if you call a method with less arguments.
    * An example would be
-   *     function foo(bar: string = "default"): string {
-   *       return bar;
-   *     }
-   *     foo(); // "default" for both AS3 and ECMAScript semantics
-   *     foo(undefined); // 'undefined' in AS3, "default" in ECMAScript semantics
+   * function foo(bar: string = "default"): string {
```
Comment (fwienber) [inline jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: Please do not reformat ASDoc code examples! Rather, there is a <pre> element missing here, but that does not have to do anything with this PR.

**Review by fwienber — CHANGES_REQUESTED** [review]

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java (hunk 3 above)
Comment (fwellers) [reply jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: This probably was the auto format in idea, i will undo it.

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java [hunk 4]
```diff
@@ -184,7 +197,8 @@
    *     }
    *     foo(); // "default" for both AS3 and ECMAScript semantics
    *     foo(undefined); // 'undefined' in AS3, "default" in ECMAScript semantics
-   */
+   *
+  */
```
Comment (fwienber) [inline jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: Almost... ;-)

**Review by fwienber — CHANGES_REQUESTED** [review]

**Pull request merged**